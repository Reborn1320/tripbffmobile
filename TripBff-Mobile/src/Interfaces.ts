//TODO: make sure screen don't necessary depend on this data
export interface ImportImageVM {
    url: string
    isSelected: boolean
}

export interface LocationDetailVM {
    long: number
    lat: number
    address: string
}

export interface LocationVM {
    location: LocationDetailVM //TODO: change to details
    images: Array<ImportImageVM>
}

export interface TripVM {
    id: number
    locations: Array<LocationVM>
}

export interface UserVM {
    username: string
    email: string
    firstName: string
    lastName: string
    fullName: string

    token: string
    //TODO: expired time...
}

//testing home screen
export interface RepoVM {
    loading?: boolean
    repos: Array<string>
    error?: string
}

export interface BffStoreData {
    repo?: RepoVM
    user?: UserVM
    trips?: Array<TripVM>
}