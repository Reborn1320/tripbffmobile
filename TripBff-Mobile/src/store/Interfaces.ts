import * as moment from 'moment';
export module StoreData {
    export interface ImportImageVM {
        imageId: string,
        url: string, //url stored in local mobile
        externalStorageId?: string,
    }

    export interface LocationDetailVM {
        long: number
        lat: number
        address: string
    }

    export interface LocationVM {
        locationId: string
        location: LocationDetailVM //TODO: simplify this by removing the interface
        fromTime: moment.Moment
        toTime: moment.Moment
        images: Array<ImportImageVM>
    }

    export interface TripVM {
        tripId: string
        name: string
        fromDate: moment.Moment
        toDate: moment.Moment
        locations: Array<LocationVM>,
        infographicId: string
    }

    export interface UserVM {
        username: string
        email: string
        firstName: string
        lastName: string
        fullName: string

        token: string,
        fbToken?: string
        //TODO: expired time...
    }

    export interface BffStoreData {
        user?: UserVM
        trips?: Array<TripVM>
    }
}


export namespace RawJsonData {
    export interface TripVM {
        tripId: string
        name: string
        fromDate: string
        toDate: string
        locations: Array<StoreData.LocationVM>,
        infographicId: string
    }

    export interface LocationVM {
        locationId: string
        location: StoreData.LocationDetailVM
        fromTime: string
        toTime: string
        images: Array<StoreData.ImportImageVM>
    }

}