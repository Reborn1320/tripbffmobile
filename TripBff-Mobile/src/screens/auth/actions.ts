import { StoreData } from '../../store/Interfaces';

export const AUTH_ADD_TOKEN = "AUTH_ADD_TOKEN"

export function addToken(user: StoreData.UserVM) {
    return {
        type: AUTH_ADD_TOKEN, user
    }
}