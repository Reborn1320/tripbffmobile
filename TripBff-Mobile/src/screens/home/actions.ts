import { ThunkResultBase } from "../_shared/LayoutContainer";

export function loginUsingUserPass(email, password): ThunkResultBase {
  return async function(dispatch, getState, extraArguments) {
    var loginUser = {
      email: email,
      password: password
    };

    return extraArguments.loginApi.post(`local/login`, loginUser);
  };
}

export function uploadSimpleImage(uri: string): ThunkResultBase {
  return async function(dispatch, getState, extraArguments) {
    return extraArguments.uploadApi
      .upload("/uploadImage", uri, { fileName: "image.jpg" })
      .then(() => {
        console.log("uploaded");
        console.log(arguments);
      })
      .catch(err => {
        console.log("err");
        console.log(err);
      });
  };
}
