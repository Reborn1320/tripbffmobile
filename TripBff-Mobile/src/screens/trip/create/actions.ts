import { StoreData } from '../../../store/Interfaces';

export const TRIP_ADD = "TRIP_ADD"

export function createTrip(trip: StoreData.TripVM) {
    return {
        type: TRIP_ADD, trip
    }
}