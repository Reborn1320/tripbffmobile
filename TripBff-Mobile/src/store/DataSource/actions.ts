import { StoreData } from "../Interfaces"

export const DataSource_GetAllFeeling = "DataSource_GetAllFeeling"

export function getAllFeelings(feelings: Array<StoreData.PreDefinedFeelingVM>) {
    return {
        type: DataSource_GetAllFeeling, feelings
    }
}

 