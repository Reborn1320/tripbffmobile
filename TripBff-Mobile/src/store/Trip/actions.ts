import { StoreData, RawJsonData } from "../Interfaces"
import moment, { Moment } from "moment";

export const LOCATION_REMOVE = "TRIP_LOCATION_REMOVE"
export const LOCATION_ADD = "TRIP_LOCATION_ADD"
export const LOCATION_UPDATE_FEELING = "TRIP_LOCATION_UPDATE_FEELING"
export const LOCATION_UPDATE_ACTIVITY = "TRIP_LOCATION_UPDATE_ACTIVITY"
export const TRIP_UPDATE_DATE_RANGE = "TRIP_UPDATE_DATE_RANGE"
export const TRIP_UPDATE_TRIP_NAME = "TRIP_UPDATE_TRIP_NAME"
export const LOCATION_UPDATE_ADDRESS = "TRIP_LOCATION_UPDATE_ADDRESS"
export const LOCATION_UPDATE_IMAGES = "TRIP_LOCATION_UPDATE_IMAGES"

export type TripActions = RemoveLocation
| UpdateTripDateRange
;

type RemoveLocation = {
    type: "TRIP_LOCATION_REMOVE",
    tripId: string
    dateIdx: number
    locationId: string
}

type UpdateTripDateRange = {
    type: "TRIP_UPDATE_DATE_RANGE",
    tripId: string
    fromDate: Moment
    toDate: Moment
    locations: StoreData.LocationVM[]
}

export function removeLocation(tripId: string, dateIdx: number, locationId: string) {
    return {
        type: LOCATION_REMOVE, tripId, dateIdx, locationId
    }
}

export function addLocation(tripId: string, dateIdx: number, location: StoreData.LocationVM) {
    return {
        type: LOCATION_ADD, tripId, dateIdx, location
    }
}


export function updateTripDateRange(tripId: string, fromDate: moment.Moment, toDate: moment.Moment, locations: StoreData.LocationVM[]) {
    return {
        type: TRIP_UPDATE_DATE_RANGE, tripId, fromDate, toDate, locations
    }
}

export function updateTripName(tripId: string, tripName: string) {
    return {
        type: TRIP_UPDATE_TRIP_NAME, tripId, tripName
    }
}

export type LocationActions = UpdateLocationFeeling
| UpdateLocationActivity
| UpdateLocationAddress
| UpdateLocationImages
;

type UpdateLocationFeeling = {
    type: "TRIP_LOCATION_UPDATE_FEELING",
    dateIdx: number,
    locationId: string,
    feeling: StoreData.FeelingVM,
}

type UpdateLocationActivity = {
    type: "TRIP_LOCATION_UPDATE_ACTIVITY",
    dateIdx: number,
    locationId: string,
    activity: StoreData.ActivityVM,
}

type UpdateLocationAddress = {
    type: "TRIP_LOCATION_UPDATE_ADDRESS",
    dateIdx: number,
    locationId: string,
    location: RawJsonData.LocationAddressVM
}


type UpdateLocationImages = {
    type: "TRIP_LOCATION_UPDATE_IMAGES",
    dateIdx: number,
    locationId: string,
    locationImages: StoreData.ImportImageVM[]
}
export function updateLocationFeeling(tripId: string, dateIdx: number, locationId: string, feeling: StoreData.FeelingVM) {
    return {
        type: LOCATION_UPDATE_FEELING, tripId, dateIdx, locationId, feeling
    }
}

export function updateLocationActivity(tripId: string, dateIdx: number, locationId: string, activity: StoreData.ActivityVM) {
    return {
        type: LOCATION_UPDATE_ACTIVITY, tripId, dateIdx, locationId, activity
    }
}
 
export function updateLocationAddress(tripId: string, dateIdx: number, locationId: string, location: RawJsonData.LocationAddressVM) {
    return {
        type: LOCATION_UPDATE_ADDRESS, tripId, dateIdx, locationId, location
    }
}

export function updateLocationImages(tripId: string, locationId: string, locationImages: StoreData.ImportImageVM[]) {
    return {
        type: LOCATION_UPDATE_IMAGES, tripId, locationId, locationImages
    }
}