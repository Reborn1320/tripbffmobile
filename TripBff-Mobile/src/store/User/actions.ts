export const AUTH_ADD_TOKEN = "AUTH_ADD_TOKEN";