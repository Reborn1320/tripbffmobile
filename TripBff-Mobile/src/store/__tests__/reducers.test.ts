import reducer from "../reducers";
import { TRIP_ADD } from "../../screens/trip/create/actions";
import moment = require("moment");
import { StoreData } from "../Interfaces";

describe('trip reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {}))
      // .toEqual({});
      .toMatchSnapshot();

  })

  it('should handle TRIP_ADD', () => {
    const startAction: { type: string, trip: StoreData.TripVM } = {
      type: TRIP_ADD,
      trip: {
        tripId: "tripId01",
        name: "name",
        fromDate: moment('2019-03-16T17:00:00.000Z'),
        toDate: moment('2019-03-23T16:59:59.999Z')
      }
    };
    var result = reducer(undefined, {});
    expect(reducer(result, startAction).trips).toMatchSnapshot();
  });


  it('should handle TRIPS_ADD', () => {
    const startAction = {
      type: "TRIPS_ADD",
      trips: [{
        tripId: "36f626c0-47cd-11e9-b7e2-efb95b29f029",
        name: "Ffggc 7f7",
        fromDate: moment("2019-03-16T17:00:00.000Z"),
        toDate: moment("2019-03-23T16:59:59.999Z"),
        locations: [{
          locationId: "c0ee8221-270e-41b9-beaa-9cbc11eb779d",
          location: {
            lat: 10.801314353942871,
            long: 106.64141845703125
          },
          fromTime: "2019-03-13T08:37:55.844Z",
          toTime: "2019-03-13T08:38:28.099Z",
          images: [{
            imageId: "829ba1a6-b9ee-4c09-8ee4-225b37c47111",
            url: "file:///storage/emulated/0/DCIM/Camera/20190313_153828.jpg",
            externalUrl: "",
            thumbnailExternalUrl: ""
          }, {
            imageId: "2ad22ba0-0434-4352-8d72-d048599dfbe7",
            url: "file:///storage/emulated/0/DCIM/Camera/20190313_153755.jpg",
            externalUrl: "",
            thumbnailExternalUrl: ""
          }],
          feeling: {},
          activity: {}
        }]
      }]
    };
    var result = reducer(undefined, {});
    expect(reducer(result, startAction).trips).toMatchSnapshot();
  });
})

