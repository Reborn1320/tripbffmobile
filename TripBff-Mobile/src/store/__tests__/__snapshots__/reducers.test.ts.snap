// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trip reducer should handle TRIP_ADD 1`] = `
Array [
  Object {
    "fromDate": "2019-03-16T17:00:00.000Z",
    "name": "name",
    "toDate": "2019-03-23T16:59:59.999Z",
    "tripId": "tripId01",
  },
]
`;

exports[`trip reducer should handle TRIPS_ADD 1`] = `
Array [
  Object {
    "dates": Array [
      Object {
        "date": "2019-03-16T17:00:00.000Z",
        "dateIdx": 1,
        "locationIds": Array [],
        "locations": Array [],
      },
      Object {
        "date": "2019-03-17T17:00:00.000Z",
        "dateIdx": 2,
        "locationIds": Array [],
        "locations": Array [],
      },
      Object {
        "date": "2019-03-18T17:00:00.000Z",
        "dateIdx": 3,
        "locationIds": Array [],
        "locations": Array [],
      },
      Object {
        "date": "2019-03-19T17:00:00.000Z",
        "dateIdx": 4,
        "locationIds": Array [],
        "locations": Array [],
      },
      Object {
        "date": "2019-03-20T17:00:00.000Z",
        "dateIdx": 5,
        "locationIds": Array [],
        "locations": Array [],
      },
      Object {
        "date": "2019-03-21T17:00:00.000Z",
        "dateIdx": 6,
        "locationIds": Array [],
        "locations": Array [],
      },
      Object {
        "date": "2019-03-22T17:00:00.000Z",
        "dateIdx": 7,
        "locationIds": Array [],
        "locations": Array [],
      },
    ],
    "fromDate": "2019-03-16T17:00:00.000Z",
    "locations": Array [
      Object {
        "activity": Object {},
        "feeling": Object {},
        "fromTime": "2019-03-13T08:37:55.844Z",
        "images": Array [
          Object {
            "externalUrl": "",
            "imageId": "829ba1a6-b9ee-4c09-8ee4-225b37c47111",
            "thumbnailExternalUrl": "",
            "url": "file:///storage/emulated/0/DCIM/Camera/20190313_153828.jpg",
          },
          Object {
            "externalUrl": "",
            "imageId": "2ad22ba0-0434-4352-8d72-d048599dfbe7",
            "thumbnailExternalUrl": "",
            "url": "file:///storage/emulated/0/DCIM/Camera/20190313_153755.jpg",
          },
        ],
        "location": Object {
          "lat": 10.801314353942871,
          "long": 106.64141845703125,
        },
        "locationId": "c0ee8221-270e-41b9-beaa-9cbc11eb779d",
        "toTime": "2019-03-13T08:38:28.099Z",
      },
    ],
    "name": "Ffggc 7f7",
    "toDate": "2019-03-23T16:59:59.999Z",
    "tripId": "36f626c0-47cd-11e9-b7e2-efb95b29f029",
  },
]
`;

exports[`trip reducer should return the initial state 1`] = `
Object {
  "dataSource": Object {},
  "trips": Array [],
  "user": Object {
    "email": "asdf@gmail.com",
    "fbToken": "",
    "firstName": "asdf",
    "fullName": "adffff",
    "lastName": "asdf",
    "token": "ASdf",
    "username": "asdf",
  },
}
`;
